File: POO JAVA/Aula14/src/aula14/Visualizacao.java
package aula14;

public class Visualizacao {

    //Atributos da Classe
    private Gafanhoto espectador;
    private Video filme;
    
    //Metodo Construtor

    public Visualizacao(Gafanhoto espectador, Video filme) {
        this.espectador = espectador;
        this.filme = filme;
        this.espectador.setTotAssistindo(this.espectador.getTotAssistindo() +1);
        this.filme.setViews(this.filme.getViews() +1);
    }
    
    //Metodos de Acessos e Modificadores
    public Gafanhoto getEspectador() {
        return espectador;
    }
    public void setEspectador(Gafanhoto espectador) {
        this.espectador = espectador;
    }
    public Video getFilme() {
        return filme;
    }
    public void setFilme(Video filme) {
        this.filme = filme;
    }

    @Override
    public String toString() {
        return "Visualizacao{" + "espectador=" + espectador + ", filme=" + filme + '}';
    }
    
    //Metodos da Classe
    public void avaliar(){
        this.filme.setAvaliação(5);
    }
    public void avaliar(int a){
        this.filme.setAvaliação(a);
    }
    public void avaliar(float b){
        int total = 0;
        if(b <= 25){
            total = 3;
        }else if(b <= 50){
            total = 5;
        }else if(b <= 75){
            total = 8;
        }else{
            total = 10;
        }
        this.filme.setAvaliação(total);
    }
}

File: POO PHP/Aula14/Visualizacao.php
<?php

require_once 'Gafanhoto.php';
require_once 'Video.php';

class Visualizacao {

    //Atributos da Classe
    private $espectador;
    private $filme;
    
    //Metodo Construtor
    public function __construct($espectador, $filme) {
        $this->espectador = $espectador;
        $this->filme = $filme;
        $this->filme->setViews($this->filme->getViews() + 1);
        $this->espectador->setTotAssistindo($this->espectador->getTotAssistindo() + 1);
    }
    
    //Metodo Acessores e Modificadores
    public function getEspectador() {
        return $this->espectador;
    }
    public function getFilme() {
        return $this->filme;
    }
    public function setEspectador($espectador): void {
        $this->espectador = $espectador;
    }
    public function setFilme($filme): void {
        $this->filme = $filme;
    }
        
    //Metodo da Classe
    public function avaliar1() {
        $this->filme->setAvaliacao(5);
    }
    public function avaliar2($int) {
        $this->filme->setAvaliacao($int);
    }
    public function avaliar3($float) {
        $a = 0;
        if($float <= 25){
            $this->$a = 25;
        }else if ($float <= 50) {
            $this->$a = 50;
        }else if($float <= 75){
            $this->$a = 75;
        } else {
            $this->$a = 100;
        }
        $this->filme->setAvaliacao($a);
    }
}
